doc/mgmt/OpenConfig_Path_Translation_Root_Expansion_HLD.md

```diff
@@ -0,0 +1,338 @@
+# OpenConfig Path Translation HLD #
+
+## Table of Contents
+- [Table of Contents](#table-of-contents)
+- [Revision](#revision)
+- [Scope](#scope)
+- [Definition/Abbreviation](#definitionsabbreviations)
+- [Overview](#overview)
+- [Requirements](#requirements)
+- [Architecture Design](#architecture-design)
+- [High Level Design](#high-level-design)
+- [SAI API](#sai-api)
+- [Configuration and Management](#configuration-and-management)
+- [Warmboot and Fastboot Design Impact](#warmboot-and-fastboot-design-impact)
+- [Memory Consumption](#memory-consumption)
+- [Restrictions/Limitations](#restrictionslimitations)
+- [Testing Requirements/Design](#testing-requirementsdesign)
+- [Open/Action items - if any](#openaction-items---if-any)
+
+### Revision
+| Rev  | Rev Date   | Author(s)          | Change Description |
+|------|------------|--------------------|--------------------|
+| v0.1 | 09/17/2024 | Neha Das (Google)   | Initial version |
+
+### Scope
+
+This document describes the high-level design to add a plugin-based path translation layer between the UMF gNMI server and external data clients to translate both requests and responses for interoperability between multiple clients and server implementations.
+
+### Definitions/Abbreviations
+
+- [gNMI](https://github.com/openconfig/gnmi) : gRPC Network Management Interface
+- API : Application Programming Interface
+- SONiC: Software for Open Networking in the Cloud
+- UMF : Unified Management Framework
+- FE: Frontend
+- BE: Backend
+
+### Overview
+
+Unified Management Framework (UMF) is a SONIC management framework which provides gNMI and other common north bound interfaces to manage configuration and status of SONIC based switches.
+
+The difference in paths and formats used by existing clients and multiple sever implementations, including UMF, can lead to interoperability issues. Some of the gaps that we have encountered include:
+1. Support for gNMI requests for config and telemetry at the Single root path which is currently not supported in UMF. Single root means the xpath `/openconfig`. UMF supports gNMI requests modeled in OpenConfig subtrees such as `/openconfig/interfaces`, `/openconfig/qos`, etc but not at the `/openconfig` root level.
```

Comment:
Are you planning to support Get/SET/Subscribe at this level? If so, do you have benchmark data on how it scales with more models and data?.